x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2397 @@
+
+```ts
+// [... file continues above ...]
+
+export const createPrebuiltRules = (rulesClient: RulesClient, rules: PrebuiltRuleAsset[]) =>
+  withSecuritySpan('createPrebuiltRules', async () => {
+    const result = await initPromisePool({
+      concurrency: MAX_RULES_TO_UPDATE_IN_PARALLEL,
+      items: rules,
+      executor: async (rule) => {
+        return createRules({
+          rulesClient,
+          params: rule,
+          isPrebuilt: true, // renamed from "immutable"
+          defaultEnabled: false,
+        });
+      },
+    });
+
+    return result;
+  });
+```
+
+- [**Review Rule Upgrade** - `POST /prebuilt_rules/upgrade/_review` (Internal)](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_upgrade/review_rule_upgrade_route.ts)
+
+This endpoint uses the `convertPrebuiltRuleAssetToRuleResponse` method to get a `RuleResponse`-type object from a target version of a rule provided from upstream. This method needs to be modified as described in the section that details the changes needed for the **Review Rule Installation** - `POST /prebuilt_rules/installation/_review`.
+
+An additional change will be needed within the route handler, when calculating the `targetRule` that is returned in the response. Since the `prebuilt` rule field is calculated by the `convertPrebuiltRuleAssetToRuleResponse` method, the `prebuilt.isCustomized` field would always be set to `false`, even when updating rules that have been previously customized. Therefore, the `prebuilt` field needs to be overwritten to take as its value for `prebuilt.isCustomized` the current value that is set in the currently installed rule, or default to false it that value doesn't yet exist:
+
+_Source: [x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_upgrade/review_rule_upgrade_route.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_upgrade/review_rule_upgrade_route.ts)_
+
+```ts
+// [... file continues above ...]
+
+const calculateRuleInfos = (results: CalculateRuleDiffResult[]): RuleUpgradeInfoForReview[] => {
+  return results.map((result) => {
+    const { ruleDiff, ruleVersions } = result;
+    const installedCurrentVersion = ruleVersions.input.current;
+    const targetVersion = ruleVersions.input.target;
+
+    // [... file continues ...]
+
+    const targetRule: RuleResponse = {
+      ...convertPrebuiltRuleAssetToRuleResponse(targetVersion),
+      // Overwrite the prebuilt object returned from convertPrebuiltRuleAssetToRuleResponse
+      // to account for the currently installed rule's value of `isCustomized`.
+      prebuilt: {
+        isCustomized: installedCurrentVersion.prebuilt?.isCustomized ?? false,
+        elasticUpdateDate: targetVersion.elasticUpdateDate,
+      },
```

Comment:
Why do we need to overwrite `prebuilt` here like that? I read the accompanying text, but didn't understand the reason.

Reply:
This doesn't make sense, we wouldn't even display this data to the user, they are techinal fields.

Remove.